### ISSUE TITLE
I wan to disable these rules jsx-a11y/anchor-has-content and jsx-a11y/anchor-is-valid

### ISSUE THREAD

**Issue by ManuelFernando** [evcohen/eslint-plugin-jsx-a11y#553]
I installed eslint-plugin-jsx-a11y over reactjs v16.8.1 and added .eslintrc file but the warning messages continue appearing

`{
  "plugins": ["jsx-a11y"],
  "rules": {
    "jsx-a11y/anchor-is-valid": "off",
    "jsx-a11y/anchor-has-content": "off"
  }
}`

**Comment by ljharb** [comment]
Where do they appear? Make sure you’re checking on the command line; if they’re appearing in your editor but not on the CLI then your editor is broken.

**Comment by ManuelFernando** [comment]
this appears on the CLI

**Comment by ljharb** [comment]
Where are the warnings appearing? What’s the exact text of the warnings? I notice you don’t have root: true, do you have an eslintrc higher up?

**Comment by ManuelFernando** [comment]
yes and .eslintrc in the root, this is my eslintrc

{
  "plugins": ["jsx-a11y"],
  "rules": {
    "jsx-a11y/anchor-is-valid": "off",
    "jsx-a11y/anchor-has-content": "off"
  }
}

I tried setting 0 instead of “off” but the result is the same

In the CLI appears this:

./src/App.js
  Line 49:  Anchors must have content and the content must be accessible by a screen reader                                                                                                                                                                                                                                                                                   jsx-a11y/anchor-has-content
  Line 49:  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

**Comment by ljharb** [comment]
Are you running a global or a local eslint? You shouldn’t even have a global eslint installed.

**Comment by ManuelFernando** [comment]
local, I have others projects with eslint and work well

**Comment by ljharb** [comment]
And are you sure this is the only eslintrc in the project? There’s no inline eslint override comments in App.js?

**Comment by ManuelFernando** [comment]
yes sure

**Comment by ljharb** [comment]
Im not really sure then :-/ it’s not possible for a rule to keep itself on when you’ve turned it off via eslint. Presumably there’s something somewhere turning it on.

What’s `eslint --print-config` print out?

**Comment by ManuelFernando** [comment]
@ljharb I even created a new app from zero with create-react-app

I got this out after executing the command

Value for 'print-config' of type 'path::String' required.

**Comment by ljharb** [comment]
CRA is different; you can’t really override the eslint rules it uses - you’d have to eject first.

**Comment by ljharb** [comment]
`eslint --print-config src/App`?

**Comment by ManuelFernando** [comment]
{
  "globals": {},
  "env": {},
  "rules": {
    "jsx-a11y/anchor-is-valid": "off",
    "jsx-a11y/anchor-has-content": "off"
  },
  "parserOptions": {},
  "plugins": [
    "jsx-a11y"
  ]
}

**Comment by ljharb** [comment]
given that, i have no idea how those rules could still be erroring. What version of eslint and eslint-plugin-jsx-a11y are you using?

**Comment by ManuelFernando** [comment]
"react": "^16.8.1”,
"eslint-plugin-jsx-a11y": "^6.2.1”

react installed eslint 5.6.0

**Comment by ljharb** [comment]
that's old; try the latest version of eslint?

**Comment by ljharb** [comment]
also note again that CRA has its own version of everything; you have to eject to override most things.

**Comment by ManuelFernando** [comment]
this wasn’t a solution because when I installed the latest version of eslint this created a problem with the dependency tree, react claims for eslint “5.6.0” :/

**Comment by ljharb** [comment]
Again, you have to eject from CRA to customize it in any meaningful way.

If you're using CRA without ejecting, you are forfeiting the ability to do most customizations, including to the eslint config. I suspect react wants `^5.6.0` tho, which includes `5.13.0`.

**Comment by ljharb** [comment]
This seems answered. If you’re using CRA, and you find yourself unable to override something, you’ll need to file an issue on CRA.

**Comment by stephanebruckert** [comment]
https://stackoverflow.com/a/47875838/1515819

**Comment by waynebloss** [comment]
Something like this could work for you:

```jsx
/** Hide props to avoid `jsx-a11y` warnings that CRA forces on you. */
const propsToHideFromEslint = {
  accessKey: "i",
};

const MyComponent = () => <button {...propsToHideFromEslint} title="(Alt + I)">I like it</button>;

```

**Comment by delasign** [comment]
I posted the answer here: https://stackoverflow.com/questions/45764602/icon-link-without-text-eslint-error

Quick access:

`const ExternalButton = styled( ({ children, ...props}) => <a {...props}>{children}</a>)`

They use of children is what'll fix it.
